### PR TITLE
Switch to pytest-variables for credentials. Fixes #768

### LINKED ISSUE #768: Switch to using pytest-variables for credentials
We should move the handling of credentials to pytest-variables. This means changing the example credentials.yml file to variables.json with suitable top level keys and migrating the tests to use the injected variables. We should also add pytest-variables as a dependency.

The variables file would probably look something like this:

```json
{
  "users": {
    "default": {
      "email": "",
      "password": "",
      "name": ""},
    "paypal": {
      "email": "",
      "password": "",
      "name": ""}
  }
}
```

An example of a migrated test would look something like this:

```python
def test_user_can_login_and_logout_using_normal_login(self, mozwebqa, variables):
    user = variables['users']['default']
    home_page = Home(mozwebqa)
    home_page.login(user['email'], user['password'])
    Assert.true(home_page.is_the_current_page)
    Assert.true(home_page.header.is_user_logged_in)
    home_page.header.click_logout()
    Assert.false(home_page.header.is_user_logged_in)
```

I notice that we currently have a `login` method that accepts a `method` argument, which can either be 'normal' or 'browserid'. It would be great to clean this up at the same time. If Browser ID (Persona) is supported then we should generate a test user and login using that. If it's not supported, then we should remove this path in the tests.

You can also see https://github.com/mozilla/sumo-tests/commit/df597f07a1a4b0d72f795bc4dc46a7f387532f29 for an example of migrating to pytest-variables.

### PR DESCRIPTION
**Pull request by davehunt**
This currently leaves the Browser ID login paths, however as far as I can tell these are never used. I would like to clean this up at the same time, but please feel free to review what I have so far.

Pinging @bobsilverberg for review and @krupa for information regarding the Browser ID paths.

### CODE REVIEW

**Comment by davehunt** [comment]
Adhoc dev: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/amo.adhoc/2/
Adhoc dev mobile: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/amo.mobile.adhoc/1/

File: pages/desktop/base.py [hunk 1]
```diff
@@ -22,20 +22,20 @@ class Base(Page):
...
     _footer_locator = (By.CSS_SELECTOR, "#footer")
 
-    def login(self, method="normal", user="default"):
+    def login(self, email, password, method="normal"):
         if method == "normal":
             login = self.header.click_login_normal()
-            login.login_user_normal(user)
+            login.login_user_normal(email, password)
 
         elif method == "browserID":
```
Comment (bobsilverberg) [inline pages/desktop/base.py]: I checked with @krupa and we can remove all of this `browserID` stuff.

File: pages/desktop/user.py [hunk 2]
```diff
@@ -27,22 +27,14 @@ class Login(Base):
...
-        email.send_keys(credentials['email'])
-
-        password = self.selenium.find_element(*self._password_locator)
-        password.send_keys(credentials['password'])
-
-        password.send_keys(Keys.RETURN)
-
-    def login_user_browser_id(self, user):
-        credentials = self.testsetup.credentials[user]
+    def login_user_browser_id(self, email, password):
```
Comment (bobsilverberg) [inline pages/desktop/user.py]: As mentioned above, we can remove this.

File: tests/desktop/test_collections.py [hunk 3]
```diff
@@ -52,24 +52,21 @@ def test_create_and_delete_collection(self, mozwebqa):
...
-        home_page.login()
+        home_page.login(existing_user['email'], existing_user['password'])
         Assert.true(home_page.is_the_current_page)
         Assert.true(home_page.header.is_user_logged_in)
 
-        username = mozwebqa.credentials['default']['name']
         my_collections_page = home_page.header.click_my_collections()
-        Assert.equal('Collections by %s :: Add-ons for Firefox' % username, home_page.page_title)
-        Assert.equal('Collections by %s' % username, my_collections_page.my_collections_header_text)
+        Assert.equal('Collections by %s :: Add-ons for Firefox' % existing_user['name'], home_page.page_title)
```
Comment (bobsilverberg) [inline tests/desktop/test_collections.py]: Nit: This line, and the next one, are awfully long. Suggest breaking onto two lines.

File: tests/desktop/test_paypal.py [hunk 4]
```diff
@@ -18,12 +18,10 @@ class TestPaypal:
     addon_name = 'Firebug'
 
     @pytest.mark.login
-    def test_that_user_can_contribute_to_an_addon(self, mozwebqa):
+    def test_that_user_can_contribute_to_an_addon(self, mozwebqa, existing_user, paypal):
```
Comment (bobsilverberg) [inline tests/desktop/test_paypal.py]: Perhaps `paypal_user` would be a bit more descriptive of a variable name than just `paypal`.

File: tests/desktop/test_paypal.py [hunk 5]
```diff
@@ -41,7 +39,7 @@ def test_that_user_can_contribute_to_an_addon(self, mozwebqa):
         Assert.true(addon_page.is_the_current_page)
 
     @pytest.mark.login
-    def test_that_user_can_make_a_contribution_without_logging_into_amo(self, mozwebqa):
+    def test_that_user_can_make_a_contribution_without_logging_into_amo(self, mozwebqa, paypal):
```
Comment (bobsilverberg) [inline tests/desktop/test_paypal.py]: As above, suggest changing to `paypal_user`.

**Comment by bobsilverberg** [comment]
LGTM. Nice work, @davehunt. r+ wc

File: pages/desktop/base.py (hunk 1 above)
Comment (davehunt) [inline pages/desktop/base.py]: :+1:

File: tests/desktop/test_collections.py (hunk 3 above)
Comment (davehunt) [inline tests/desktop/test_collections.py]: :+1:

File: tests/desktop/test_paypal.py (hunk 4 above)
Comment (davehunt) [inline tests/desktop/test_paypal.py]: :+1:

**Comment by davehunt** [comment]
Removed BrowserID and addressed feedback comments.
Results: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/amo.variables.adhoc/4/

**Comment by davehunt** [comment]
I missed a BrowserID reference, which was found by the adhoc build. It's fixed in a new commit and a new build is running: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/amo.variables.adhoc/5/

**Comment by davehunt** [comment]
Sorry, one more :blush: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/amo.variables.adhoc/6/

**Comment by bobsilverberg** [comment]
@davehunt This is all fine for me now. All tests pass locally. r+

**Comment by davehunt** [comment]
Merged as https://github.com/mozilla/Addon-Tests/commit/b2b8122732a881bdc43614c6a17c60d3f2f59832

**Pull request merged**